mantle/kola/tests/misc/network.go

```diff
@@ -219,6 +228,67 @@ var (
           umount /var/mnt
 	`
 
+	setupVethPairsTemplate = `#!/usr/bin/env bash
+          set -ex
+
+          create_veth_pair() {
+            veth_host_side_end_name=$1
+            veth_netns_side_end_name=$2
+            host_side_mac_address=$3
+            netns_side_ip=$4
+            network_namespace=$5
+
+            # Create veth pair and assign a namespace to veth-netns
+            ip link add ${veth_host_side_end_name} type veth peer name ${veth_netns_side_end_name}
+            ip link set ${veth_netns_side_end_name} netns ${network_namespace}
+
+            # Assign an MAC address to the host-side veth end
+            ip link set dev ${veth_host_side_end_name} address ${host_side_mac_address}
+
+            # Assign an IP address to netns-side veth end and bring it up
+            ip netns exec ${network_namespace} ip address add ${netns_side_ip} dev ${veth_netns_side_end_name}
+            ip netns exec ${network_namespace} ip -6 addr flush ${veth_netns_side_end_name}
+            ip netns exec ${network_namespace} ip link set ${veth_netns_side_end_name} up
+          }
+
+          activate_veth_end() {
+            veth_end_name=$1
+            nmcli dev set ${veth_end_name} managed yes
+            ip link set ${veth_end_name} up
```

Comment:
If you use nmcli above, why not here?

Reply:
well I didn't find a one liner to replace this line.
```
$ nmcli con s
NAME                UUID                                  TYPE      DEVICE     
Wired connection 1  b7e6f4f5-6893-3ad4-bcd5-663cecfff7a7  ethernet  ens5       
Wired connection 2  81c099c1-ff96-3d2d-aa01-0c9db86c48ac  ethernet  veth1-host 
Wired connection 3  1f3d8a11-9aeb-3165-b3a8-7b879db21972  ethernet  --         
[core@cosa-devsh ~]$ nmcli con up 'Wired connection 3'
Error: Connection activation failed: No suitable device found for this connection (device ens5 not available because profile is not compatible with device (mismatching interface name)).
```

Reply:
I guess you are leveraging the side-effect that when a NIC is managed and brought UP, NM will create a profile for it. IMO it would be cleaner if you just created the profile explicitly and stuck to nmcli commands instead of mixing them with ip link.

Reply:
I agree. I think the most straight forward way is `sudo nmcli con up id 'Wired connection 3'` but that's giving me an error I don't understand (see above)..
I am not finding the correct command to do that. Can you hint me to the right way to do it?

Reply:
Hey, I think that the reason I'm getting this error is because network manager does not fully support veths in this version
IIUC, nmcli supports veth only from [network-manager 1.30](https://networkmanager.dev/blog/networkmanager-1-30/)
which AFAIK is used only on rhel 8.4 (our current rhcos image is forked from 8.3)
```
[core@cosa-devsh ~]$ nmcli --version
nmcli tool, version 1.26.0-14.1.rhaos4.7.el8
[core@cosa-devsh ~]$ cat /etc/redhat-release 
Red Hat Enterprise Linux CoreOS release 4.7
[core@cosa-devsh ~]$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux CoreOS"
VERSION="47.83.202105220305-0"
VERSION_ID="4.7"
OPENSHIFT_VERSION="4.7"
RHEL_VERSION="8.3"
PRETTY_NAME="Red Hat Enterprise Linux CoreOS 47.83.202105220305-0 (Ootpa)"
ID="rhcos"
ID_LIKE="rhel fedora"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::coreos"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="OpenShift Container Platform"
REDHAT_BUGZILLA_PRODUCT_VERSION="4.7"
REDHAT_SUPPORT_PRODUCT="OpenShift Container Platform"
REDHAT_SUPPORT_PRODUCT_VERSION="4.7"
OSTREE_VERSION='47.83.202105220305-0'
```

Reply:
Sound good